server/test/unit/paTest/handleMessageClientsTest.js

```diff
@@ -0,0 +1,226 @@
+      sandbox.stub(twilioHelpers, 'sendTwilioMessage').callsFake(mockSendTwilioMessage)
+
+      this.res = mockResponse(sandbox)
+      await pa.handleMessageClients(
+        {
+          body: { twilioMessage: testTwilioMessage, googleIdToken: testGoogleIdToken },
+        },
+        this.res,
+      )
+    })
+
+    it('should respond with status 200', () => {
+      expect(this.res.status).to.be.calledWith(200)
+    })
+
+    it('should respond with the Twilio message as twilioMessage', () => {
+      expect(this.res.body.twilioMessage).to.equal(testTwilioMessage)
+    })
+
+    it('should respond with the expected phone numbers in successfullyMessaged', () => {
+      const toPhoneNumbers = this.res.body.successfullyMessaged.map(twilioTraceObject => twilioTraceObject.to)
+
+      // deep equal; no other phone numbers should be contacted
+      expect(toPhoneNumbers).to.eql(successfulPhoneNumbers)
+    })
+
+    it('should respond with the expected phone numbers in failedToMessage', () => {
+      const toPhoneNumbers = this.res.body.failedToMessage.map(twilioTraceObject => {
+        return twilioTraceObject.to
+      })
+
+      // deep equal; no other phone numbers should fail
+      expect(toPhoneNumbers).to.eql(failPhoneNumbers)
+    })
+
+    it('should call twilioHelpers.sendTwilioMessage for all active clients', () => {
+      // expect twilioHelpers.sendTwilioMessage to be called with each of the expected arguments
+      expectedArguments.forEach(args => {
+        expect(twilioHelpers.sendTwilioMessage).to.be.calledWith(args.toPhoneNumber, args.fromPhoneNumber, args.twilioMessage)
+      })
+    })
+  })
+
+  describe('for a request when Twilio is not operating as expected', () => {
+    beforeEach(async () => {
+      sandbox.stub(twilioHelpers, 'sendTwilioMessage').returns(undefined)
+
+      this.res = mockResponse(sandbox)
+      await pa.handleMessageClients(
+        {
+          body: { twilioMessage: testTwilioMessage, googleIdToken: testGoogleIdToken },
+        },
+        this.res,
+      )
+    })
+
+    it('should respond with status 200', () => {
+      expect(this.res.status).to.be.calledWith(200)
+    })
+
+    it('should respond with the Twilio message as twilioMessage', () => {
+      expect(this.res.body.twilioMessage).to.equal(testTwilioMessage)
+    })
+
+    it('should respond with no phone numbers in successfullyMessaged', () => {
+      expect(this.res.body.successfullyMessaged).to.eql([])
+    })
+
+    it('should respond with all phone numbers in failedToMessage', () => {
+      // create array of all phone numbers (successful and fail)
+      const allPhoneNumbers = [...successfulPhoneNumbers, ...failPhoneNumbers]
+
+      // create array of to phone numbers
+      const toPhoneNumbers = this.res.body.failedToMessage.map(twilioTraceObject => twilioTraceObject.to)
+
+      // expect all (failed) to phone numbers to be equal to all phone numbers
+      expect(toPhoneNumbers).to.eql(allPhoneNumbers)
```

Comment:
I appreciate your comments in this, this makes understanding your logic so much easier!